### ISSUE TITLE
experiment.save is exposed in the python API

### ISSUE THREAD

**Issue by andreasjansson** [replicate/keepsake#64]
I was tired and accidentally wrote `experiment.save` rather than `experiment.commit`. My linter didn't complain because `.save()` a method on Experiment. The error I got was

```
Traceback (most recent call last):
  File "train.py", line 145, in <module>
    train()
  File "train.py", line 140, in train
    train_loop(model, train_dl, test_dl)
  File "train.py", line 116, in train_loop
    experiment.save(
TypeError: save() got an unexpected keyword argument 'step'
```

Might be worth prefixing "private" methods with underscore.

**Comment by enjoythecode** [comment]
Hi, I'd like to take on this issue as my first open source contribution.

Just to make sure, my understanding is that this involves changing the name of `save` under Experiment to `_save` in its definition and everywhere it is called. Is that correct?

Thanks

**Comment by andreasjansson** [comment]
Hi @enjoythecode, that's exactly right! There are another couple of methods on the `Experiment` class that are unnecessarily public:
* `save()`
* `validate()`
* `start_heartbeat()`

Really appreciate your contribution!

**Comment by bfirsh** [comment]
Fixed in #542
